app/poros/week.rb

```diff
@@ -0,0 +1,26 @@
+class Week
+  def self.determine(event_data)
+    found_event = get_valid_week(split_events(event_data))
+    get_week_number(found_event)
+  end
+
+  private
+
+    def self.split_events(event_data)
+      event_data.map{ |event| event[:summary].split }
+    end
+
+    def self.get_valid_week(summary_arrays)
+      summary_arrays.find { |split_event| valid_week(split_event) }
+    end
+
+    def self.valid_week(split_event)
+      first = split_event.first.downcase
+      split_event.length == 2 && (first == 'week' || first == 'intermission')
+    end
+
+    def self.get_week_number(event)
+      return -1 if event.nil?
```

Comment:
not 100% this covers all edge cases, `event` should be nil if `get_valid_week` returns nil:
1. the API query returned no results (in which case it returns an empty array)
2. the condition in line 19 fails for every event